### PR TITLE
During gauge creation return error instead of panic.

### PR DESCRIPTION
**Pull request by rghetia**

### CODE REVIEW

File: metric/gauge.go [hunk 1]
```diff
@@ -24,19 +25,19 @@ import (
 	"go.opencensus.io/metric/metricdata"
 )
 
-// gauge represents a quantity that can go up an down, for example queue depth
+// gauge represents a quantity that can go up an down, for example queueInt64 depth
```
Comment (songy23) [inline metric/gauge.go]: > queueInt64

Typo?

File: metric/error_const.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+const (
+	errGaugeExistsWithDiffType = "gauge with same name exists with a different type"
```
Comment (songy23) [inline metric/error_const.go]: You can use error constants instead:
```go
var (
  errGaugeExistsWithDiffType = errors.New("gauge with same name exists with a different type")
 ...
)
```

Comment (rghetia) [reply metric/error_const.go]: SG.

File: metric/gauge.go (hunk 1 above)
Comment (rghetia) [reply metric/gauge.go]: yes. I'll fix it.

**Pull request merged**